src/librustc/ty/layout.rs

```diff
@@ -1614,6 +1624,13 @@ impl<'a, 'tcx, C> TyLayoutMethods<'tcx, C> for Ty<'tcx>
                 this.ty.simd_type(tcx)
             }
 
+            ty::TyAdt(def, ..) if Some(def.did) == tcx.lang_items().va_list() => {
```

Comment:
The `VaList` implementation in `core` is just a stub. When compiling for `X86_64`, any access to `field` will result in an out of bounds index.